src/qibocal/calibrations/characterization/qubit_spectroscopy.py

```diff
@@ -412,3 +412,150 @@ def qubit_spectroscopy_flux(
+        points (int): Save data results in a file every number of points
+
+    Returns:
+        - A DataUnits object with the raw data obtained for the fast and precision sweeps with the following keys
+
+            - **MSR[V]**: Resonator signal voltage mesurement in volts
+            - **i[V]**: Resonator signal voltage mesurement for the component I in volts
+            - **q[V]**: Resonator signal voltage mesurement for the component Q in volts
+            - **phase[rad]**: Resonator signal phase mesurement in radians
+            - **frequency[Hz]**: Qubit drive frequency value in Hz
+            - **current[A]**: Current value in A applied to the flux line
+            - **qubit**: The qubit being tested
+            - **fluxline**: The fluxline being tested
+            - **iteration**: The iteration number of the many determined by software_averages
+
+        - A DataUnits object with the fitted data obtained with the following keys
+
+            - **qubit_freq**: frequency
+            - **peak_voltage**: peak voltage
+            - **popt0**: Lorentzian's amplitude
+            - **popt1**: Lorentzian's center
+            - **popt2**: Lorentzian's sigma
+            - **popt3**: Lorentzian's offset
+            - **qubit**: The qubit being tested
+    """
+    platform.reload_settings()
+
+    sequence = PulseSequence()
+    ro_pulses = {}
+    qd_pulses = {}
+    for qubit in qubits:
+        qd_pulses[qubit] = platform.create_qubit_drive_pulse(
+            qubit, start=0, duration=5000
+        )
+        ro_pulses[qubit] = platform.create_qubit_readout_pulse(qubit, start=5000)
+        sequence.add(qd_pulses[qubit])
+        sequence.add(ro_pulses[qubit])
+
+    # define the parameter to sweep and its range:
+    # qubit drive frequency
+    qubit_frequencies = {}
+    for qubit in qubits:
+        qubit_frequencies[qubit] = platform.characterization["single_qubit"][qubit][
+            "qubit_freq"
+        ]
+    delta_frequency_range = np.arange(-freq_width // 2, freq_width // 2, freq_step)
+
+    # flux current
+    sweetspot_currents = {}
+    for qubit in qubits:
+        sweetspot_currents[qubit] = platform.characterization["single_qubit"][qubit][
+            "sweetspot"
+        ]
+    current_range = np.arange(-current_width // 2, current_width // 2, current_step)
+
+    # create a DataUnits object to store the results,
+    # DataUnits stores by default MSR, phase, i, q
+    # additionally include qubit frequency and flux current
+    data = DataUnits(
+        name=f"data",
+        quantities={"frequency": "Hz", "current": "A"},
+        options=["qubit", "fluxline", "iteration"],
+    )
+
+    count = 0
+    center = {}
+    for iteration in range(software_averages):
+
+        for current in current_range:
+            for qubit in qubits:
+                # set new flux current
+                platform.qb_port[qubit].current = current + sweetspot_currents[qubit]
```

Comment:
In qibolab main we have now a `platform.set_current(qubit, curr)` method
```suggestion
            for qubit in qubits:
                # set new flux current
                platform.set_current(qubit, current) = current + qubits[qubit].sweetspot
```